### PR TITLE
feat: Add Advanced settings in new Advanced Tab

### LINKED ISSUE #12611: [Wallet - Token management] Add `Advanced settings` in new `Advanced Tab`
[Design](https://www.figma.com/file/eM26pyHZUeAwMLviaS1KJn/%E2%9A%99%EF%B8%8F-Wallet-Settings%3A-Manage-Tokens?type=design&node-id=305%3A139866&mode=design&t=HBhmgzvmu47bpf8o-1):

Screenshot 2023-12-19 at 14 33 55

### LINKED ISSUE #13040: [Tokens Management] Create new `Advanced tab` and move `Tokens List tab` content to it

### PR DESCRIPTION
**Pull request by caybro**
... under Settings/Wallet/Manage Tokens

- rename Tokens List tab to Advanced tab
- introduce a new `CurrencyAmountInput` component, backed by `FormattedDoubleProperty` C++ class (plus the respective SB page and a QML test case)
- use `FastExpressionFoo` for the collectibles views as well

Fixes #12611
Fixes #13040

NOTE: This needs integration with the backend (done separately)

### Affected areas

Settings/Wallet/Manage tokens

### Screenshot of functionality (including design for comparison)

- [x] I've checked the design and this PR matches it

image

SB for CurrencyAmountInput:
image

### CODE REVIEW

File: storybook/pages/CurrencyAmountInputPage.qml [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+        SplitView.fillHeight: true
+
+        ColumnLayout {
+            Layout.fillWidth: true
+            RowLayout {
+                Label {
+                    text: "Value:\t"
+                }
+                TextField {
```
Comment (micieslak) [inline storybook/pages/CurrencyAmountInputPage.qml:42]: Using `CurrencyAmountInput` component causes breaking binding with this text field and changing content here does nothing.

File: storybook/pages/CurrencyAmountInputPage.qml [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+                    font.bold: true
+                    text: input.valid ? "true" : "false"
+                }
+            }
+            RowLayout {
+                Layout.fillWidth: true
+                Label {
+                    text: "Formatted as locale string:"
+                }
+                Label {
```
Comment (micieslak) [inline storybook/pages/CurrencyAmountInputPage.qml:111]: It actually doesn't show `"Formatted as locale string:"` but just the text from the component:

Screenshot from 2024-01-08 22-28-25

File: ui/StatusQ/include/StatusQ/formatteddoubleproperty.h [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    void valueChanged();
+
+private:
+    QVariant value() const;
+    void setValue(const QVariant& newValue);
+    double m_value{0.0};
+
+    QString valueAsString() const;
+
+    QLocale m_locale{QLocale()};
```
[the remark this answers is not in the document]
Comment (micieslak) [reply ui/StatusQ/include/StatusQ/formatteddoubleproperty.h:27]: I would be beneficial to expose locale via a property. When it's not accessible the component is limited to default locale and is also not easily testable.

File: ui/StatusQ/include/StatusQ/formatteddoubleproperty.h [hunk 4]
```diff
@@ -0,0 +1,28 @@
+#pragma once
+
+#include <QmlTypeAndRevisionsRegistration>
+
+class FormattedDoubleProperty : public QObject
```
Comment (micieslak) [inline ui/StatusQ/include/StatusQ/formatteddoubleproperty.h:5]: Imo this component requires some basic documentation describing what's the purpose and behavior of that component.
At least at first glance that component does the work that can be easily done in qml without instantiating additional QObject for that (var property for value and readonly string property for parsed string representation).

Comment (micieslak) [reply ui/StatusQ/include/StatusQ/formatteddoubleproperty.h:5]: Unit tests should be added. They would also nicely show corner cases behavior.

File: ui/imports/shared/controls/CurrencyAmountInput.qml [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+    readonly property string valueAsString: internalProp.valueAsString
+
+    FormattedDoubleProperty {
+        id: internalProp
+        onValueChanged: root.text = asString() // min number of decimals, strip zeroes
+    }
+
+    Keys.onPressed: (event) => {
+                        // additionally accept dot (.) and convert it to the correct decimal point char
+                        if (event.key === Qt.Key_Period) {
```
Comment (micieslak) [inline ui/imports/shared/controls/CurrencyAmountInput.qml:31]: Should't `Key_Comma` be handled in the same way? For some locale period is the decimal char.

File: storybook/pages/CurrencyAmountInputPage.qml [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+                text: "Enabled"
+                checked: true
+            }
+            RowLayout {
+                Layout.fillWidth: true
+                Layout.topMargin: 16
+                Label {
+                    text: "Numeric value:"
+                }
+                Label {
```
Comment (micieslak) [inline storybook/pages/CurrencyAmountInputPage.qml:85]: It seems to not update the entered value properly:

Screenshot from 2024-01-08 22-52-26

**Review by micieslak — COMMENTED** [review]
Looks good in general, just some questions/doubts.

File: ui/StatusQ/include/StatusQ/formatteddoubleproperty.h (hunk 3 above)
[the remark this answers is not in the document]
Comment (caybro) [reply ui/StatusQ/include/StatusQ/formatteddoubleproperty.h:27]: Sure, I can do that

File: ui/StatusQ/include/StatusQ/formatteddoubleproperty.h [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+public:
+    explicit FormattedDoubleProperty(QObject* parent = nullptr);
+
+    Q_INVOKABLE QString asString(int decimals = QLocale::FloatingPointShortest) const;
+
+signals:
+    void valueChanged();
+
+private:
+    QVariant value() const;
```
[the remark this answers is not in the document]
Comment (caybro) [reply ui/StatusQ/include/StatusQ/formatteddoubleproperty.h:21]: To be able to accept a value as a string

File: ui/StatusQ/include/StatusQ/formatteddoubleproperty.h [hunk 8]
```diff
@@ -0,0 +1,28 @@
+#pragma once
+
+#include <QmlTypeAndRevisionsRegistration>
```
[the remark this answers is not in the document]
Comment (caybro) [reply ui/StatusQ/include/StatusQ/formatteddoubleproperty.h:3]: Ah, it's for the automatic type registration using the `QML_ELEMENT` macro below (which we don't make use of yet); I can remove that; it's more of a future-proof solution :)

File: ui/imports/shared/controls/CurrencyAmountInput.qml (hunk 5 above)
Comment (caybro) [reply ui/imports/shared/controls/CurrencyAmountInput.qml:31]: Then that would be the locale's decimal point and we don't need to do anything there

File: storybook/pages/CurrencyAmountInputPage.qml (hunk 6 above)
Comment (caybro) [reply storybook/pages/CurrencyAmountInputPage.qml:85]: You need to press ENTER/RETURN or shift the focus away to confirm that value

File: ui/StatusQ/include/StatusQ/formatteddoubleproperty.h (hunk 4 above)
Comment (caybro) [reply ui/StatusQ/include/StatusQ/formatteddoubleproperty.h:5]: Agree, I'll add something

File: storybook/pages/CurrencyAmountInputPage.qml (hunk 1 above)
Comment (caybro) [reply storybook/pages/CurrencyAmountInputPage.qml:42]: It works here both ways, if you press ENTER or TAB away

File: storybook/pages/CurrencyAmountInputPage.qml (hunk 2 above)
Comment (caybro) [reply storybook/pages/CurrencyAmountInputPage.qml:111]: Right... it should probably show `NaN` if the input (internal value) is invalid

File: storybook/pages/CurrencyAmountInputPage.qml (hunk 1 above)
Comment (caybro) [reply storybook/pages/CurrencyAmountInputPage.qml:42]: This happens when the internal validator gets into this broken state; no idea what's causing this or how to prevent it yet

File: storybook/pages/CurrencyAmountInputPage.qml (hunk 6 above)
Comment (caybro) [reply storybook/pages/CurrencyAmountInputPage.qml:85]: I changed that to format on the fly

File: ui/StatusQ/include/StatusQ/formatteddoubleproperty.h (hunk 3 above)
[the remark this answers is not in the document]
Comment (caybro) [reply ui/StatusQ/include/StatusQ/formatteddoubleproperty.h:41]: Done

File: storybook/qmlTests/tests/tst_CurrencyAmountInput.qml [hunk 9]
```diff
@@ -0,0 +1,168 @@
...
+            id: input
+            anchors.centerIn: parent
+        }
+    }
+
+    // SignalSpy {
+    //     id: notificationSpy
+    //     target: Global
+    //     signalName: "displayToastMessage"
+    // }
```
Comment (micieslak) [inline storybook/qmlTests/tests/tst_CurrencyAmountInput.qml:24]:
```suggestion
```

File: storybook/qmlTests/tests/tst_CurrencyAmountInput.qml [hunk 10]
```diff
@@ -0,0 +1,168 @@
...
+    TestCase {
+        name: "CurrencyAmountInput"
+        when: windowShown
+
+        property CurrencyAmountInput controlUnderTest: null
+
+        function init() {
+            controlUnderTest = createTemporaryObject(componentUnderTest, root)
+            controlUnderTest.locale = "en_US"
+            //notificationSpy.clear()
```
Comment (micieslak) [inline storybook/qmlTests/tests/tst_CurrencyAmountInput.qml:42]:
```suggestion
```

**Review by micieslak — COMMENTED** [review]
One more unexpected behavior found:

[Kazam_screencast_00078.webm](https://github.com/status-im/status-desktop/assets/20650004/b25d24ad-4f66-4f78-aefd-1754ccfb068c)

1. Cursor positioned on the beginning
2. `8` pressed

File: storybook/qmlTests/tests/tst_CurrencyAmountInput.qml (hunk 9 above)
Comment (caybro) [reply storybook/qmlTests/tests/tst_CurrencyAmountInput.qml]: Removed

File: storybook/qmlTests/tests/tst_CurrencyAmountInput.qml (hunk 10 above)
Comment (caybro) [reply storybook/qmlTests/tests/tst_CurrencyAmountInput.qml]: Removed

**Comment by micieslak** [comment]
> > One more unexpected behavior found:
> > Kazam_screencast_00078.webm
> > 
> > 1. Cursor positioned on the beginning
> > 2. `8` pressed
> 
> I find that logical; you have `0.1` at the beginning, you prepend `8` and you end up with `80.1`

Sorry, I didn't state it explicitly. The problem is with jumping cursor here. It jumps unexpectedly to the end.

**Review by micieslak — COMMENTED** [review]
There is still a problem with handling locale. For me and potentially other people depending on their settings the tests fail:
```
FAIL!  : QmlTests::CurrencyAmountInput::test_decimals() 'verify()' returned FALSE. ()
   Loc: [/home/michalc/workspace/status-desktop/storybook/qmlTests/tests/tst_CurrencyAmountInput.qml(96)]
PASS   : QmlTests::CurrencyAmountInput::test_explicitLocale()
FAIL!  : QmlTests::CurrencyAmountInput::test_getSetValueProgramatically() 'verify()' returned FALSE. ()
   Loc: [/home/michalc/workspace/status-desktop/storybook/qmlTests/tests/tst_CurrencyAmountInput.qml(51)]
FAIL!  : QmlTests::CurrencyAmountInput::test_inputValueManually() 'verify()' returned FALSE. ()
   Loc: [/home/michalc/workspace/status-desktop/storybook/qmlTests/tests/tst_CurrencyAmountInput.qml(77)]
FAIL!  : QmlTests::CurrencyAmountInput::test_validator() 'verify()' returned FALSE. ()
   Loc: [/home/michalc/workspace/status-desktop/storybook/qmlTests/tests/tst_CurrencyAmountInput.qml(166)]
```
The problem is probably in both `FormattedDoubleProperty` and `DoubleValidator` bc that handle locales by name.

User's system locale may be e.g. `en_US` but with decimal separator `,`. So changing locale by name to `en_US` will take no effect bc it's the same string, decimal point won't be changed to value default for `en_US`.

File: ui/StatusQ/CMakeLists.txt [hunk 11]
```diff
@@ -110,6 +110,7 @@ add_library(StatusQ SHARED
         include/StatusQ/submodelproxymodel.h
         include/StatusQ/sumaggregator.h
         include/StatusQ/writableproxymodel.h
+        include/StatusQ/formatteddoubleproperty.h
```
Comment (alexjba) [inline ui/StatusQ/CMakeLists.txt:113]: Whould be nice to keep them sorted

**Review by alexjba — COMMENTED** [review]
Nice work! The code looks good to me!

Some minor questions:

1. What's the expected behaviour of the currency amount input when the decimal separator is added after the numeric input. E.g: Set the number of decimals to 2. Type 1234567. Type the decimal separator after `1` (1,12). Currently the user input is ignored and a decimal separator cannot be added.

2. Sometimes when removing the decimal separator the number is altered

https://github.com/status-im/status-desktop/assets/47811206/36e2808c-fb2c-493c-b5a8-87f85755edc1

**Comment by alexjba** [comment]
> There is still a problem with handling locale. For me and potentially other people depending on their settings the tests fail:
> 
> ```
> FAIL!  : QmlTests::CurrencyAmountInput::test_decimals() 'verify()' returned FALSE. ()
>    Loc: [/home/michalc/workspace/status-desktop/storybook/qmlTests/tests/tst_CurrencyAmountInput.qml(96)]
> PASS   : QmlTests::CurrencyAmountInput::test_explicitLocale()
> FAIL!  : QmlTests::CurrencyAmountInput::test_getSetValueProgramatically() 'verify()' returned FALSE. ()
>    Loc: [/home/michalc/workspace/status-desktop/storybook/qmlTests/tests/tst_CurrencyAmountInput.qml(51)]
> FAIL!  : QmlTests::CurrencyAmountInput::test_inputValueManually() 'verify()' returned FALSE. ()
>    Loc: [/home/michalc/workspace/status-desktop/storybook/qmlTests/tests/tst_CurrencyAmountInput.qml(77)]
> FAIL!  : QmlTests::CurrencyAmountInput::test_validator() 'verify()' returned FALSE. ()
>    Loc: [/home/michalc/workspace/status-desktop/storybook/qmlTests/tests/tst_CurrencyAmountInput.qml(166)]
> ```
> 
> The problem is probably in both `FormattedDoubleProperty` and `DoubleValidator` bc that handle locales by name.
> 
> User's system locale may be e.g. `en_US` but with decimal separator `,`. So changing locale by name to `en_US` will take no effect bc it's the same string, decimal point won't be changed to value default for `en_US`.
> 
> Probably the easiest workaround is to pinpoint defaults by adding:
> 
> ```
> QLocale::setDefault(QLocale(QLocale::English, QLocale::UnitedStates));
> ```
> 
> in qml tests' `main.cpp`.

The tests are failing on me as well. Although I see the locale is set in the test case `init` function.

File: ui/StatusQ/CMakeLists.txt (hunk 11 above)
Comment (micieslak) [reply ui/StatusQ/CMakeLists.txt:113]: Alt+Shift+S in Qtc :)

File: ui/imports/shared/controls/CurrencyAmountInput.qml [hunk 12]
```diff
@@ -0,0 +1,120 @@
...
+                            root.insert(root.cursorPosition, Qt.locale(root.locale).decimalPoint)
+                            event.accepted = true
+                        } else if (event.key >= Qt.Key_A && event.key <= Qt.Key_Z) { // reject typing non-numbers (can happen when the validator is in an intermediate state
+                            event.accepted = true
+                        }
```
[the remark this answers is not in the document]
Comment (caybro) [reply ui/imports/shared/controls/CurrencyAmountInput.qml:56]: Yup, I can do

**Comment by caybro** [comment]
> Nice work! The code looks good to me!
> 
> Some minor questions:
> 
> 1. What's the expected behaviour of the currency amount input when the decimal separator is added after the numeric input. E.g: Set the number of decimals to 2. Type 1234567. Type the decimal separator after `1` (1,12). Currently the user input is ignored and a decimal separator cannot be added.

Yeah this is correct, the QML validator takes the number of decimals into account, so it prevents you from typing it at that place

> 2. Sometimes when removing the decimal separator the number is altered

I!ll have a look

File: ui/app/AppLayouts/Profile/views/wallet/ManageTokensView.qml [hunk 13]
```diff
@@ -128,10 +133,74 @@ ColumnLayout {
...
+                Layout.fillWidth: true
+                Layout.topMargin: 40 + 18
+                Layout.bottomMargin: 26
+                text: qsTr("Asset settings")
+                color: Theme.palette.baseColor1
+            }
+            StatusDialogDivider {
+                Layout.fillWidth: true
+            }
+            StatusListItem {
```
Comment (noeliaSD) [inline ui/app/AppLayouts/Profile/views/wallet/ManageTokensView.qml:162]: From the design, it's not clear to me that these 2 switches options should be displayed as selectable list elements. At least, probably the background should not be changed when hovering. I mean that:
Screenshot 2024-01-11 at 10 18 15

File: ui/app/AppLayouts/Profile/views/wallet/ManageTokensView.qml [hunk 14]
```diff
@@ -128,10 +133,74 @@ ColumnLayout {
     }
 
     Component {
-        id: supportedTokensListPanel
-        SupportedTokenListsPanel {
-            sourcesOfTokensModel: root.sourcesOfTokensModel
-            tokensListModel: root.tokensListModel
+        id: advancedTab
+        ColumnLayout {
```
Comment (noeliaSD) [inline ui/app/AppLayouts/Profile/views/wallet/ManageTokensView.qml:137]: I've just seen in the design that it's needed to add the list of tokens that are hidden bc they have low balance.  [Design](https://www.figma.com/file/eM26pyHZUeAwMLviaS1KJn/%E2%9A%99%EF%B8%8F-Wallet-Settings%3A-Manage-Tokens?type=design&node-id=305%3A142331&mode=design&t=81rOxWBKpouduNoM-1). Here a screenshot:
Screenshot 2024-01-11 at 10 16 01

File: ui/app/AppLayouts/Profile/views/wallet/ManageTokensView.qml [hunk 15]
```diff
@@ -128,10 +133,74 @@ ColumnLayout {
...
+            spacing: 0
+            StatusBaseText {
+                Layout.fillWidth: true
+                Layout.topMargin: 18
+                Layout.bottomMargin: 18
+                text: qsTr("Token lists")
+                color: Theme.palette.baseColor1
+            }
+            SupportedTokenListsPanel {
```
Comment (noeliaSD) [inline ui/app/AppLayouts/Profile/views/wallet/ManageTokensView.qml:146]: It looks like the layout of the token list delegate has changed slightly. It's not related to this pr but would be nice if you also add the changes in here since they are really small changes:

1. Subtitle text is now 13px
2. Button `View` doesn't have background by default.

Screenshot 2024-01-11 at 10 12 08

File: ui/imports/shared/controls/CurrencyAmountInput.qml (hunk 12 above)
[the remark this answers is not in the document]
Comment (caybro) [reply ui/imports/shared/controls/CurrencyAmountInput.qml]: Fixed

File: ui/StatusQ/CMakeLists.txt (hunk 11 above)
Comment (caybro) [reply ui/StatusQ/CMakeLists.txt]: Fixed

**Review by noeliaSD — COMMENTED** [review]
Quick first review after app inspection: some design comments added.

I'll continue with code inspection!

File: ui/app/AppLayouts/Profile/views/wallet/ManageTokensView.qml (hunk 14 above)
Comment (caybro) [reply ui/app/AppLayouts/Profile/views/wallet/ManageTokensView.qml:137]: OK, that's in the new design; I'd leave that for later

File: ui/app/AppLayouts/Profile/views/wallet/ManageTokensView.qml (hunk 13 above)
Comment (caybro) [reply ui/app/AppLayouts/Profile/views/wallet/ManageTokensView.qml:162]: Yeah this is kinda unclear from the design what the color should be when hovering; it's unfortunate that the input bg color is the same as the list item's hover color in our palette :/ Something for @benjthayer to decide?

File: ui/app/AppLayouts/Profile/views/wallet/ManageTokensView.qml (hunk 15 above)
Comment (caybro) [reply ui/app/AppLayouts/Profile/views/wallet/ManageTokensView.qml:146]: OK, I can do that

Comment (caybro) [reply ui/app/AppLayouts/Profile/views/wallet/ManageTokensView.qml:146]: Fixed

image

File: ui/StatusQ/src/formatteddoubleproperty.cpp [hunk 16]
```diff
@@ -0,0 +1,66 @@
...
+
+QVariant FormattedDoubleProperty::value() const
+{
+    return m_value;
+}
+
+void FormattedDoubleProperty::setValue(const QVariant& newValue)
+{
+    if (!newValue.isValid()) {
+        qWarning() << "Setting property to invalid value:" << newValue << "is not supported";
```
Comment (noeliaSD) [inline ui/StatusQ/src/formatteddoubleproperty.cpp:19]: Suggestion: Maybe you could provide more explicit waring info by providing the `.typeName()` value.

File: ui/StatusQ/src/formatteddoubleproperty.cpp [hunk 17]
```diff
@@ -0,0 +1,66 @@
...
+        return;
+    }
+
+    auto ok = false;
+    auto tempValue = newValue.toDouble(&ok);
+
+    if (!ok && newValue.canConvert<QString>()) {
```
Comment (noeliaSD) [inline ui/StatusQ/src/formatteddoubleproperty.cpp:26]: Probably not a big deal but is it necessary to convert to string and back to double if `newValue` is already a double type?

File: ui/app/AppLayouts/Profile/views/wallet/ManageTokensView.qml (hunk 14 above)
Comment (noeliaSD) [reply ui/app/AppLayouts/Profile/views/wallet/ManageTokensView.qml:137]: Ok! Could you create subtask for that?

File: ui/app/AppLayouts/Profile/views/wallet/ManageTokensView.qml [hunk 18]
```diff
@@ -128,10 +133,74 @@ ColumnLayout {
...
+            }
+            StatusListItem {
+                Layout.fillWidth: true
+                title: qsTr("Don’t display assets with balance lower than")
+
+                components: [
+                    CurrencyAmountInput {
+                        enabled: displayThresholdSwitch.checked
+                        currencySymbol: SharedStores.RootStore.currencyStore.currentCurrency
+                        value: 0.10
```
Comment (noeliaSD) [inline ui/app/AppLayouts/Profile/views/wallet/ManageTokensView.qml:190]: It could be a `d` readonly property!

File: ui/imports/shared/controls/CurrencyAmountInput.qml [hunk 19]
```diff
@@ -0,0 +1,121 @@
...
+        internalProp.value
+        return root.valid ? internalProp.asLocaleString(root.decimals) : "NaN"
+    }
+    readonly property string asString: internalProp.asString // "C" locale string
+
+    FormattedDoubleProperty {
+        id: internalProp
+        onValueChanged: {
+            const oldPos = root.cursorPosition
+            root.text = asLocaleString() // min number of decimals, strip zeroes
```
Comment (noeliaSD) [inline ui/imports/shared/controls/CurrencyAmountInput.qml:44]: ? Not sure :)
```suggestion
            root.text = asLocaleString() // min number of decimals, strip zeros
```

**Review by noeliaSD — COMMENTED** [review]
Code LGTM! Just added minor suggestions

File: ui/app/AppLayouts/Profile/views/wallet/ManageTokensView.qml (hunk 14 above)
Comment (caybro) [reply ui/app/AppLayouts/Profile/views/wallet/ManageTokensView.qml:137]: Done here: https://github.com/status-im/status-desktop/issues/13183

File: ui/StatusQ/src/formatteddoubleproperty.cpp (hunk 17 above)
Comment (caybro) [reply ui/StatusQ/src/formatteddoubleproperty.cpp:26]: We are not converting _to_ string, but _from_ string and iff the conversion from a `double` fails

File: ui/app/AppLayouts/Profile/views/wallet/ManageTokensView.qml (hunk 18 above)
Comment (caybro) [reply ui/app/AppLayouts/Profile/views/wallet/ManageTokensView.qml:190]: Could be if you insist :)

File: ui/imports/shared/controls/CurrencyAmountInput.qml (hunk 19 above)
Comment (caybro) [reply ui/imports/shared/controls/CurrencyAmountInput.qml:44]: Bot hare correct actually :)

File: storybook/qmlTests/tests/tst_CurrencyAmountInput.qml [hunk 20]
```diff
@@ -0,0 +1,182 @@
...
+
+    TestCase {
+        name: "CurrencyAmountInput"
+        when: windowShown
+
+        property CurrencyAmountInput controlUnderTest: null
+
+        function init() {
+            controlUnderTest = createTemporaryObject(componentUnderTest, root)
+            controlUnderTest.locale = "en_US.utf8"
```
Comment (micieslak) [inline storybook/qmlTests/tests/tst_CurrencyAmountInput.qml:35]:
```suggestion
            controlUnderTest.locale = "en_US"
```
the `.utf8` part is ignored from what I know so it may be confusing why it's here.

File: ui/app/AppLayouts/Profile/views/wallet/ManageTokensView.qml (hunk 18 above)
Comment (caybro) [reply ui/app/AppLayouts/Profile/views/wallet/ManageTokensView.qml:190]: Ah on a second thought, what exactly you meant here?

**Review by micieslak — COMMENTED** [review]
Tests pass nicely now :). Looks good for me but there are some valid change requests pointed by others.

File: storybook/qmlTests/tests/tst_CurrencyAmountInput.qml (hunk 20 above)
Comment (caybro) [reply storybook/qmlTests/tests/tst_CurrencyAmountInput.qml]: Fixed

File: ui/StatusQ/src/formatteddoubleproperty.cpp (hunk 16 above)
Comment (caybro) [reply ui/StatusQ/src/formatteddoubleproperty.cpp:19]: Fixed

File: ui/app/AppLayouts/Profile/views/wallet/ManageTokensView.qml (hunk 13 above)
Comment (caybro) [reply ui/app/AppLayouts/Profile/views/wallet/ManageTokensView.qml:164]: Also, as they are standard `StatusListItem`s they ought to default to have the hover enabled (just like anywhere else in the Settings)

**Review by noeliaSD — APPROVED** [review]
Thanks for the changes!

**Review by micieslak — APPROVED** [review]
